### ISSUE TITLE
Sensors not working when MAC address characters written in lower case

### ISSUE THREAD

**Issue by theliquid** [custom-components/ble_monitor#734]
Hi,
I just discovered that my Xiaomi BLE sensors don't work, when I write the characters in the MAC address in the configuration.yaml file in lower case.
Since writing the MAC address in lower case is a common practice, this seems to be a bug.

**Comment by Ernst79** [comment]
Can confirm the bug. It’s also happening in the UI configuration. When I change the upper case MAC to a lower case MAC, It will create additional entities with _2 at the end in the unique ID (sensor name is lower case) but it won’t update these, it will keep on updating the old ones with the upper case names.

**Comment by Ernst79** [comment]
Hmm, if I add a new sensor with lower case MAC, is does seem to works OK. Looking further....

**Comment by Ernst79** [comment]
Now, I tried with YAML. 

```yaml
ble_monitor:
  hci_interface: 0
  bt_auto_restart: False
  active_scan: False
  discovery: False
  period: 60
  use_median: False
  decimals: 1
  log_spikes: False
  restore_state: False
  report_unknown: False
  devices:
    # sensor
    - mac: 'a4:c3:38:5d:18:b4'
      name: 'Livingroom'
      encryption_key: 'b9ea895fac7eea6d30532432a5123456'
      temperature_unit: C
      decimals: 2
      use_median: False
      restore_state: default
```

It's working fine. 

![image](https://user-images.githubusercontent.com/24375847/154859577-b726d505-5887-4770-bbd1-f9ff714a5ee0.png)

Could you show me your configuration?

**Comment by Ernst79** [comment]
Ok, when I remove the name from my YAML config, I see the issue. It seems to initially create the correct entities:
- lower case sensor name: `ble temperature a4c1388d18b2`
- entity_id: `sensor.ble_temperature_a4c1388d18b2`

Later on in the process, it creates a second entity of each sensor
- upper case sensor name: `ble temperature A4C1388D18B2`
- entity_id: `sensor.ble_temperature_a4c1388d18b2_2` (note the added `_2`)

![image](https://user-images.githubusercontent.com/24375847/154860095-6120d6f2-3843-4ec4-bff0-edb6de1b0eb3.png)

The last one is the one that is being updated with data.

**Comment by Ernst79** [comment]
I can see the same in the logs.

First with lower case letters

```
2022-02-20 20:04:50 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble temperature a4c1388d18b2
2022-02-20 20:04:50 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble humidity a4c1388d18b2
2022-02-20 20:04:50 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble battery a4c1388d18b2
2022-02-20 20:04:50 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble voltage a4c1388d18b2
2022-02-20 20:04:50 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble rssi a4c1388d18b2
```

and 5 seconds later with upper case letters

```
2022-02-20 20:04:55 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble temperature A4C1388D18B2
2022-02-20 20:04:55 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble humidity A4C1388D18B2
2022-02-20 20:04:55 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble battery A4C1388D18B2
2022-02-20 20:04:55 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble voltage A4C1388D18B2
2022-02-20 20:04:55 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble rssi A4C1388D18B2
```

**Comment by Ernst79** [comment]
I tried it a second time, and now it only adds the sensor with uppercase letters, with exactly the same config.

**Comment by Ernst79** [comment]
Fixed in beta version 7.9.0-beta. Could you check if it works for you?

**Comment by theliquid** [comment]
Works for me with the beta version! Thanks!

**Comment by Ernst79** [comment]
Fixed as final in 7.9.2
